### ISSUE TITLE
Cant remove Public Read from _User Object during a sign up.

### ISSUE THREAD

**Issue by gateway** [parse-community/parse-php-sdk#389]
I need the ability to create a parse user from our backend admin tool and set proper ACL for only that user to be able to read/write to his or her object. 

Our mobile client can set the ACL to the users object id created, this from what we understand only allows the user to be able to do any changes to the users object.

On the mobile client this is how we set it.

![image](https://user-images.githubusercontent.com/519563/36499909-81debc18-16f7-11e8-910b-03538bb8755a.png)

Doing almost the similar process with the PHP sdk doesnt seem to work.

```
  // Signup
  $user = new ParseUser();
  $user->setUsername($data['email']);
  $user->setPassword($data['password']);
  $user->setEmail($data['email']);
  try {
    $user->signUp();
    // deal with ACL stuff after user is created.
    $acl = new ParseACL($user);
    $acl->setPublicReadAccess(false);
    $acl->setPublicWriteAccess(false);
    //$acl->setUserWriteAccess($user, true);
    //$acl->setUserReadAccess($user, true);
    $user->save();
    $user->logOut();
```

Looking at the sdk it doesnt seem like you dan do this.. 

```
    $acl->setPublicReadAccess($user, false);
    $acl->setPublicWriteAccess($user, false);
```
Which the our mobile clients are doing..

Basically with the code above creates a row like this in our database.

![image](https://user-images.githubusercontent.com/519563/36500238-9b4b2a6e-16f8-11e8-8553-390e72042497.png)

Issue, or am I missing some sort of other function call to fix this?

**Comment by gateway** [comment]
After a bit of testing I found the solution and I'm posting it here just incase anyone runs into this..

```
$user = new ParseUser();
$user->setUsername("foo");
$user->setPassword("Q2w#4!o)df");
try {
    $user->signUp();
    $acl = ParseACL::createACLWithUser($user);
    $user->setACL($acl);
    $user->save();
} catch (ParseException $ex) {
    // error in $ex->getMessage();
}
```

the key being these two lines after the $user->signUp();

```
    $acl = ParseACL::createACLWithUser($user);
    $user->setACL($acl);
```
